### PR TITLE
Update link to `#fileID` documentation that has moved to the stdlib.

### PR DESCRIPTION
**Pull request by grynspan**
`#fileID` is no longer a literal but is now a macro in the stdlib, so its documentation has moved. Update our documentation to match.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

**Review by stmontgomery — APPROVED** [review]
Seems fine but I wonder if we could avoid it by phrasing as "…in the documentation for `#fileID` in the standard…"

**Comment by grynspan** [comment]
> Seems fine but I wonder if we could avoid it by phrasing as "…in the documentation for `#fileID` in the standard…"

Got an alternative? :)

File: Sources/Testing/SourceAttribution/SourceLocation.swift [hunk 1]
```diff
@@ -32,11 +31,14 @@ public struct SourceLocation: Sendable {
   /// instance's ``fileID`` property is `"FoodTruck/WheelTests.swift"`, the
   /// file name is `"WheelTests.swift"`.
   ///
+  /// The structure of file IDs is described in the documentation for the
+  /// [`#fileID`](https://developer.apple.com/documentation/swift/fileID())
+  /// macro in the Swift standard library.
```
Comment (stmontgomery) [inline Sources/Testing/SourceAttribution/SourceLocation.swift:36]:
```suggestion
  /// The structure of file IDs is described in the documentation for
  /// [`#fileID`](https://developer.apple.com/documentation/swift/fileID())
  /// in the Swift standard library.
```

**Comment by grynspan** [comment]
This is a comments-only change. Merging.

**Pull request merged**